File: src/main/java/de/todesstoss/tsreports/util/reflection/ReflectionUtils.java
package de.todesstoss.tsreports.util.reflection;

import org.jetbrains.annotations.NotNull;

import java.lang.reflect.Field;

public class ReflectionUtils {

    public static <T> void setField(@NotNull Object object,
                                    @NotNull String fieldName,
                                    @NotNull T value) {
        try {
            Field field = object.getClass().getDeclaredField(fieldName);
            boolean accessible = field.isAccessible();

            field.setAccessible(true);
            field.set(object, value);
            field.setAccessible(accessible);
        } catch (Exception e) {
            throw new NullPointerException(e.getMessage());
        }
    }

}

File: src/main/java/de/todesstoss/tsreports/util/reflection/ReflectionUtils.kt
package de.todesstoss.tsreports.util.reflection

object ReflectionUtils {

    fun <T> setField(
        obj: Any,
        fieldName: String,
        value: T
    ) {
        try {
            val field = obj.javaClass.getDeclaredField( fieldName )
            val accessible = field.isAccessible

            field.isAccessible = true
            field.set( obj, value )
            field.isAccessible = accessible
        } catch (ex: Exception) {
            throw NullPointerException( ex.message )
        }
    }

}